src/kibana/plugins/dashboard/components/modal/modal.js

```diff
@@ -0,0 +1,122 @@
+                                 $scope.ejs.RangeFilter('TimeUpdated')
+                                    .from(from)
+                                    .to(to)
+                              )
+                           )
+                        )
+                        .doSearch()
+                        .then(
+                           function(result) {
+                              $scope.savableRule.loading = false;
+
+                              if (result && result.facets && result.facets.rules && result.facets.rules.count !== undefined) {
+                                 $scope.savableRule.numMatches = result.facets.rules.count;
+                              } else {
+                                 $scope.savableRule.state = 'error';
+                                 $scope.savableRule.error = 'There was a problem executing your search.';
+                              }
+                           },
+                           function() {
+                              $scope.savableRule.loading = false;
+                              $scope.savableRule.state = 'error';
+                              $scope.savableRule.error = 'There was a problem executing your search.';
+                           }
+                        );
+                  },
+                  function() {
+                     $scope.savableRule.loading = false;
+                     $scope.savableRule.state = 'error';
+                     $scope.savableRule.error = 'There was a problem executing your search.';
+                  }
+               );
+         };
+
+        $scope.saveRule = function(rule) {
+            rule.query = $scope.elasticSearchFields.convertQuery(rule.query);
+
+            var query = $scope.ejs.QueryStringQuery(rule.query);
+
+            rule.loading = true;
+           Restangular
+              .all('rules/')
+              .post({
+                 name: rule.name,
+                 enabled: true,
+                 severity: rule.severity,
+                 query: angular.fromJson(query.toString())
+              })
+              .then(
+                 function(data) {
+                   if (!data.error){
+                     $scope.savableRule = angular.copy(data);
+                     $scope.savableRule.state = 'saved';
+                     $scope.savableRule.loading = false;
+                   } else{
+                     $scope.savableRule = {
+                           loading: false,
+                           state: 'error',
+                           error: data.message
+                     };
+                   }
+                 },
+                function() {
+                   $scope.savableRule = {
+                      loading: false,
+                      state: 'error',
+                      error: 'There was a problem saving your rule.'
+                   };
```

Comment:
A little too much duplication for the error handling to make me happy but I can see why it's easier to make it like this than doing reuse